app/main.py

```
@@ -1,44 +1,64 @@
+# app/main.py
+import asyncio
 import time
+from typing import Awaitable, Any
 
 from iot.devices import HueLightDevice, SmartSpeakerDevice, SmartToiletDevice
 from iot.message import Message, MessageType
 from iot.service import IOTService
 
 
-def main() -> None:
-    # create an IOT service
+async def run_sequence(*functions: Awaitable[Any]) -> None:
+    """Run coroutines in sequence, one after another."""
+    for function in functions:
+        await function
+
+
+async def run_parallel(*functions: Awaitable[Any]) -> None:
+    """Run coroutines concurrently."""
+    await asyncio.gather(*functions, return_exceptions=True)
```

`run_parallel` uses `return_exceptions=True`. That will convert exceptions from tasks into returned exception objects rather than letting them propagate. This can hide failures during registration/message sending. Consider removing `return_exceptions=True` so exceptions propagate, or explicitly handle returned exceptions after `asyncio.gather` if partial failures are acceptable.